grazer/src/main/java/edu/asu/diging/grazer/web/PersonController.java

```diff
@@ -68,8 +70,22 @@ public int compare(WikidataStatement o1, WikidataStatement o2) {
     public String getPersonGraph(@PathVariable("personId") String personId, Model model) {
         
         IConcept concept = cache.getConceptById(personId);
+        List<Graph> graph;
+        if(concept.getType() != null && concept.getType().getUri().equals("http://www.digitalhps.org/types/TYPE_986a7cc9-c0c1-4720-b344-853f08c136ab")) {
+            graph = graphDbConnector.getGraphs(concept.getUri());
+        } else {
+            graph = graphDbConnector.getNonPeopleGraphs(concept.getUri());
+            for(Graph g: graph) {
+                List<Node> nodeList = g.getNodes();
+                List<Edge> edgeList = g.getEdges();
+                for(Node node: nodeList) {
+                    if(!node.getUri().equals(concept.getUri())) {
+                        edgeList.removeIf(Edge -> Edge.getTarget().equals(node.getId()));
+                    }
+                }
+            }
+        }
```

Comment:
a couple of things:
- the method should always do the same thing, there is no reason to distinguish between people and other concepts
- instead of iterating over all nodes and then over all edges, just iterate over all edges and check both target and source
- also, if you would check against the type, it should not be a hardcoded string! (and i'm pretty sure it's somewhere in the configs already, I hope at least)
- since personId can now be any kind of concept id, it should be renamed

Reply:
an even better approach would be to only get the edges from the database that have the concept as source or target node